### PR TITLE
FLEXIN-737: Update logic for filename truncation

### PR DESCRIPTION
**Pull request by mdeshpande12**
- Updated definition for Notification interface
- Updated logic for shortening for filename for it to truncate the filename only when filename characters are more than 20 characters
- Moved file related logic to single `file.ts` utility file.

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: cypress.config.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ export default defineConfig({
         supportFile: "cypress/support/e2e.ts",
         downloadsFolder: "cypress/downloads",
         trashAssetsBeforeRuns: true,
-        chromeWebSecurity: false,
+        chromeWebSecurity: true,
```
Comment (deepak-thomas-t) [inline cypress.config.ts:27]: is this needed?

File: src/__tests__/notifications.test.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { notifications } from "../notifications";
+describe("notifications", () => {
+    it("should shorten filename if the filename char is too long", () => {
+        const longFileName = "Profile-20230724T08574.txt";
+        const truncatedName = notifications.shortenFileName(longFileName);
```
Comment (deepak-thomas-t) [inline src/__tests__/notifications.test.ts:5]: can you add a test for a file name with 20 character and 19 character

File: cypress.config.ts (hunk 1 above)
Comment (mdeshpande12) [reply cypress.config.ts:27]: this is not needed here. I just cherry picked this commit and added it to this branch, just to make sure that the e2e tests are running fine. This commit will go away once this PR merges -> https://github.com/twilio/twilio-webchat-react-app/pull/133

File: src/__tests__/notifications.test.ts (hunk 2 above)
Comment (mdeshpande12) [reply src/__tests__/notifications.test.ts:5]: That's a good catch, the logic should be upto 20 characters. I'll change the comparison from 'less than' to 'less than or equal to 20'. Also, I think test for 19 characters is covered with the short file name test.

**Review by deepak-thomas-t — DISMISSED** [review]

File: src/notifications.ts [hunk 3]
```diff
@@ -9,10 +9,18 @@ import { Notification } from "./store/definitions";
...
+const shortenFileName = (name: string, maxChar = 20) => {
+    const [, filename, fileExtension] = name.match(/^(.+)(\.[\S]*)$/) || [];
+    if (!filename) {
+        return name;
+    }
 
-    return `${filename.substr(0, maxChar)}[...]${fileExtension || ""}`;
+    if (filename.length <= maxChar) return name;
+
+    return `${filename.substring(0, 14)}[...]${filename.substring(
```
Comment (AnudeepChPaul) [inline src/notifications.ts:20]: Is 14 the recommended by PM or anyone ?

Else I'll suggest to reduce it further. like 7

Comment (AnudeepChPaul) [reply src/notifications.ts:20]: Also, take this variable into a top level declaration.

like at top of the file,

`const MAX_DISPLAYED_CHAR = 7`

and at line 20

`return `${filename.substring(0, MAX_DISPLAYED_CHAR)}`

Comment (mdeshpande12) [reply src/notifications.ts:20]: @AnudeepChPaul yes this is recommended & confirmed by Stu and Shilpa as well.

Comment (AnudeepChPaul) [reply src/notifications.ts:20]: Then move this 14 to top level declaration. 

```const MAX_DISPLAYED_CHAR = 14 ```

at line 20,
```return ${filename.substring(0, MAX_DISPLAYED_CHAR)}```

Comment (mdeshpande12) [reply src/notifications.ts]: pushed a commit for this.

File: src/notifications.ts [hunk 4]
```diff
@@ -8,11 +8,20 @@ import { Notification } from "./store/definitions";
...
+        return name;
+    }
 
-    return `${filename.substr(0, maxChar)}[...]${fileExtension || ""}`;
+    if (filename.length <= maxChar) return name;
+
+    return `${filename.substring(0, MAX_DISPLAYED_CHAR_FOR_FIRST_PART)}[...]${filename.substring(
+        filename.length - 6,
+        filename.length - 1
+    )}${fileExtension}`;
```
Comment (deepak-thomas-t) [inline src/notifications.ts:24]: wouldn't `filename.substring(-5)` be enough instead of `filename.substring(
        filename.length - 6,
        filename.length - 1
    )`

Comment (mdeshpande12) [reply src/notifications.ts:24]: somehow this file got skipped in the commit. Actually, `filename.substring(filename.length-5)` is the shortform that would work.

File: src/notifications.ts [hunk 5]
```diff
@@ -8,11 +8,20 @@ import { Notification } from "./store/definitions";
  *     type: "neutral"
  * };
  */
+const MAX_DISPLAYED_CHAR_FOR_FIRST_PART = 14;
+const MAX_DISPLAYED_CHAR_FOR_SECOND_PART = 5;
 
-const shortenFileName = (string: string, maxChar = 50) => {
```
Comment (deepak-thomas-t) [inline src/notifications.ts:12]: This function is not related to notification right ?? It would be better move it to `src/utils`

Comment (mdeshpande12) [reply src/notifications.ts:12]: pushed a commit for this

**Comment by mdeshpande12** [comment]
Added following changes:

- Moved file related logic to single file.ts utility file.
- Reverted back the logic from commit [a7a2dc127300e8a42f21d967ce55bfaec74212d5](https://github.com/twilio/twilio-webchat-react-app/pull/133/commits/a7a2dc127300e8a42f21d967ce55bfaec74212d5); as I noticed issues in parsing of Map DataStructure in localStorage

**Review by deepak-thomas-t — APPROVED** [review]
LGTM

**Review by deepak-thomas-t — APPROVED** [review]

**Review by AnudeepChPaul — APPROVED** [review]
